delfin/drivers/fujitsu/eternus/eternus_stor.py

```diff
@@ -352,3 +391,548 @@ def parse_alert(context, alert):
+                "native_storage_host_initiator_id": host_iscsi.get('iqn'),
+                "wwn": host_iscsi.get('iqn'),
+                "status": constants.InitiatorStatus.ONLINE if
+                state is not None and state in
+                consts.HOST_PATH_STATUS_SPECIFIC_TWO else
+                constants.InitiatorStatus.OFFLINE,
+                "native_storage_host_id": iscsi_name,
+                'type': constants.InitiatorType.ISCSI,
+                'alias': host_iscsi.get('alias')
+            }
+            initiator_list.append(initiator_item)
+        host_sas_list = self.get_data(consts.GET_HOST_SAS_ADDRESSES,
+                                      consts.HOST_SAS_FIVE)
+        for host_sas in (host_sas_list or []):
+            if len(host_sas) < consts.HOST_SAS_FIVE:
+                continue
+            sas_name = host_sas[consts.HOST_SAS_ONE]
+            sas_address = host_sas[consts.HOST_SAS_TWO]
+            state = host_status.get(sas_name)
+            initiator_item = {
+                "name": sas_address,
+                "storage_id": self.storage_id,
+                "native_storage_host_initiator_id": sas_address,
+                "wwn": sas_address,
+                "status": constants.InitiatorStatus.ONLINE if
+                state is not None and state in
+                consts.HOST_PATH_STATUS_SPECIFIC_TWO else
+                constants.InitiatorStatus.OFFLINE,
+                "native_storage_host_id": sas_name,
+                'type': constants.InitiatorType.SAS
+            }
+            initiator_list.append(initiator_item)
+        return initiator_list
+
+    def list_storage_hosts(self, ctx):
+        host_list = []
+        host_status = self.get_host_status()
+        host_fc_list = self.get_data(consts.GET_HOST_WWN_NAMES,
+                                     consts.HOST_TOTAL)
+        for host_fc in (host_fc_list or []):
+            if len(host_fc) < consts.HOST_TOTAL:
+                continue
+            fc_name = host_fc[consts.HOST_NAME_COUNT]
+            os = host_fc[consts.HOST_TYPE_COUNT].lower()
+            state = host_status.get(fc_name)
+            host_d = {
+                "name": fc_name,
+                "storage_id": self.storage_id,
+                "native_storage_host_id": fc_name,
+                "os_type": consts.HOST_OS_TYPES_MAP.get(
+                    os, constants.HostOSTypes.UNKNOWN),
+                "status": constants.HostStatus.NORMAL if
+                state is not None and state in
+                consts.HOST_PATH_STATUS_SPECIFIC_TWO else
+                constants.HostStatus.OFFLINE
+            }
+            host_list.append(host_d)
+
+        host_iscsi_list = self.get_iscsi_host()
+        for host_iscsi in (host_iscsi_list or []):
+            iscsi_name = host_iscsi.get('name')
+            state = host_status.get(iscsi_name)
+            os = host_iscsi.get('os')
+            os = os.lower() if os else None
```

Comment:
It is recommended that this operation be processed in self.get_iscsi_host() and then returned.
![image](https://user-images.githubusercontent.com/59107077/153789331-51831879-dee2-401d-a665-08624ca1ff2b.png)

Reply:
get_iscsi_host is a public function. List_storage_host_initiators and list_storage_hosts are both used. If you put this code after get_iscsi_host,The function will no longer be a public function, resulting in duplicate code